scripts/tf-tests/run-jobs.py

```diff
@@ -0,0 +1,219 @@
+    except subprocess.CalledProcessError as e:
+        logging.error("Error submitting job for %s: %s", canonical_id, e.stderr)
+    return None
+
+
+def save_job_uuid(job_uuid: str, output_dir: Path, canonical_id: str):
+    """Creates a directory for the Canonical ID and saves the job UUID."""
+    id_dir = output_dir / canonical_id
+    id_dir.mkdir(exist_ok=True)
+    (id_dir / "tf_job_id.txt").write_text(job_uuid)
+    logging.info("Job submitted for %s with job_id: %s", canonical_id, job_uuid)
+
+
+def submit_all_jobs(
+    machines_file: Path, template_file: Path, output_dir: Path
+) -> Dict[str, Path]:
+    """Submit all jobs for the given machines."""
+    canonical_ids = load_canonical_ids(machines_file)
+    job_ids = {}
+
+    for canonical_id in canonical_ids:
+        job_yaml = create_job_yaml(template_file, canonical_id)
+        temp_job_file = write_temp_job_file(job_yaml, output_dir, canonical_id)
+        job_uuid = submit_job(temp_job_file, canonical_id)
+
+        if job_uuid:
+            save_job_uuid(job_uuid, output_dir, canonical_id)
+            job_ids[job_uuid] = output_dir / canonical_id
+
+        temp_job_file.unlink()  # Clean up temporary YAML file
+
+    return job_ids
+
+
+# Job Monitoring Functions
+def load_jobs(output_dir: Path) -> Dict[str, Path]:
+    """Load job IDs and directories from the job outputs directory."""
+    jobs = {}
+    for id_dir in output_dir.iterdir():
+        if id_dir.is_dir():
+            job_id_file = id_dir / "tf_job_id.txt"
+            if job_id_file.exists():
+                job_id = job_id_file.read_text().strip()
+                jobs[job_id] = id_dir
+    return jobs
+
+
+def check_job_status(job_id: str) -> Optional[str]:
+    """Check the status of a job by its job ID."""
+    try:
+        result = subprocess.run(
+            ["testflinger", "status", job_id],
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        return result.stdout.strip()
+    except subprocess.CalledProcessError as e:
+        logging.error("Error checking status for job %s: %s", job_id, e.stderr)
+    return None
+
+
+def extract_status_from_output(test_output):
+    """Extracts the status value from the test output JSON."""
+    match_ = re.search(r'"status":\s*"([^"]+)"', test_output)
+    return match_.group(1) if match_ else "Unknown"
+
+
+def retrieve_job_results(job_id: str, id_dir: Path):
+    """Retrieve and save the results of a completed job."""
+    try:
+        results_result = subprocess.run(
+            ["testflinger-cli", "results", job_id],
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        results_data = json.loads(results_result.stdout)
+        test_output = results_data.get("test_output", "")
+        (id_dir / "output.txt").write_text(test_output)
+        status = extract_status_from_output(test_output)
```

Comment:
I am a little confused here: can't we get the exit status of the test phase in the same way we get it's output, i.e.  `status = results_data.get("test_status")`?